### PR TITLE
Services and Resources

### PR DESCRIPTION
**Pull request by andrewcchau**
Separated resource and services files into appropriate sub-packages

### CODE REVIEW

File: src/main/java/lithium/university/services/TwitterServices.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    private TwitterServices() {}
+
+    public static TwitterServices getInstance() { return INSTANCE; }
+
+    /*
+     * Takes a message and error checks it before attempting to post to user's twitter
+     * Input: Twitter twitter - twitter instance, String message - message to be posted, int tweetTotal - total limited characters
+     * Output: status object of updated status if successful update, null otherwise
+     * */
+    public Status postToTwitter(Twitter twitter, String message, int tweetTotal) throws TwitterException {
```
Comment (kev-chan) [inline src/main/java/lithium/university/services/TwitterServices.java]: i like the change.  Now what happens if someone else was to pass in a long tweet vs empty tweet?

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/main/java/lithium/university/services/TwitterServices.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import twitter4j.Paging;
+import twitter4j.Status;
+import twitter4j.Twitter;
+import twitter4j.TwitterException;
+import twitter4j.TwitterFactory;
+import twitter4j.conf.ConfigurationBuilder;
+
+import java.util.List;
+
+public class TwitterServices {
```
Comment (kev-chan) [inline src/main/java/lithium/university/services/TwitterServices.java]: TwitterService is a better name

File: src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 3]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (kev-chan) [inline src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: what is this used for?

Comment (andrewcchau) [reply src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: Forgot to remove this from commit. Will take out as it's not needed with current code base

File: src/main/java/lithium/university/resources/TwitterResource.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            logger.debug("User passed in null message form data");
+            return Response.serverError().entity(getMessageFormError()).build();
+        }
+
+        /*Attempt to post to Twitter and get error codes*/
+        try{
+            twitterService.postToTwitter(twitter, message, TwitterApplication.TWEET_LENGTH);
+        }catch(TwitterException te){
+            logger.error("An exception has occurred in postTweet", te);
+            if (te.getMessage().contains("length")) {
```
Comment (kev-chan) [inline src/main/java/lithium/university/resources/TwitterResource.java]: did you think about the scenario where Twitter4j itself returns back an exception with type "length"?

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/main/java/lithium/university/resources/TwitterResource.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+
+    @POST
+    @Path("/tweet")
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    public Response postTweet(@FormParam("message") String message){
+        this.getTwitterAuthentication();
+
+        /*Initial checking of message for correct form*/
+        if(message == null){
```
Comment (kev-chan) [inline src/main/java/lithium/university/resources/TwitterResource.java]: any reason why the null check is here?

**Review by kev-chan — CHANGES_REQUESTED** [review]

**Pull request merged**